addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/ParamGuessResult.java

```diff
@@ -65,6 +65,11 @@ public List<Reason> getReasons() {
         return reasons;
     }
 
+    public HistoryReference getHistoryReference() {
+        this.historyReference.setNote(this.toString());
```

Comment:
Better do this after creating the history reference, getters should not have side effects (usually).